operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/DefaultEventHandler.java

```diff
@@ -59,27 +74,14 @@ public DefaultEventHandler(
             runnable -> new Thread(runnable, "EventHandler-" + relatedControllerName));
   }
 
-  public DefaultEventHandler(
-      CustomResourceCache customResourceCache,
-      EventDispatcher eventDispatcher,
-      String relatedControllerName,
-      Retry retry) {
-    this(
-        customResourceCache,
-        eventDispatcher,
-        relatedControllerName,
-        retry,
-        ConfigurationService.DEFAULT_RECONCILIATION_THREADS_NUMBER);
-  }
-
   @Override
   public void close() {
-    if (eventSourceManager != null) {
-      log.debug("Closing EventSourceManager {} -> {}", controllerName, eventSourceManager);
-      eventSourceManager.close();
+    try {
+      log.debug("Closing handler for {}", controllerName);
+      executor.awaitTermination(10, TimeUnit.SECONDS);
```

Comment:
This should be configurable by the user (with a default). Since we want to gracefully shutdown, so wait ideally all threads are finished.  There can be a meaningful timeout, but that depends on the implementation of the controller.

Reply:
Agreed. I still have other changes I want to make but I want first to get that PR merged so that we can proceed step by step.

Reply:
Could we move this change to that PR?

Reply:
kk, that is fine by me, until we make sure we don't forget about it :)

Reply:
https://github.com/java-operator-sdk/java-operator-sdk/issues/421